muspinsim/mpi.py

```diff
@@ -73,10 +73,8 @@ def broadcast_object(self, obj, selected_attrs=None):
             v = obj.__dict__.get(k, None)
             try:
                 v = self.comm.bcast(v, root=0)
-            except OverflowError as e:
-                raise OverflowError(
-                    "Overflow error on: {0}, {1}: {2}".format(k, v, str(e))
-                )
+            except OverflowError as exc:
+                raise OverflowError(f"Overflow error on: {k}, {v}: {str(exc)}") from exc
```

Comment:
```suggestion
                raise OverflowError(f"Overflow error on: {k}, {v}") from exc
```
Fully support the move to `raise ... from ...`, I'm just wondering if we still need to include the message from original exception since it will be in the traceback thanks to `from` anyway? Obviously the following is just a toy example with a short stack but it illustrates the difference:
```python
>>> exc = IndexError("This was the original error") 
>>> raise IndexError(f"Overflow error on some values a, b: {str(exc)}") from exc   
IndexError: This was the original error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: Overflow error on some values a, b: This was the original error
>>> raise IndexError(f"Overflow error on some values a, b") from exc
IndexError: This was the original error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: Overflow error on some values a, b
```

What are your thoughts?

Reply:
Yes totally agree, that is what its supposed to be for.